### PR TITLE
Questionable changes

### PR DESCRIPTION
**Pull request by blp**
The changes in these commits make the code easier to understand, I think, but the one for `exert` in particular might indicate that there was a bug in the original code.

### CODE REVIEW

File: crates/dbsp/src/trace/mod.rs [hunk 1]
```diff
@@ -157,7 +156,7 @@ pub trait Trace: BatchReader {
     fn recede_to(&mut self, frontier: &Self::Time);
 
     /// Exert merge effort, even without updates.
-    fn exert(&mut self, effort: &mut isize);
+    fn exert(&mut self, effort: isize);
```
Comment (Kixiron) [inline crates/dbsp/src/trace/mod.rs:159]: The issue here is that the information about how much work was done is lost where before we could exert effort on multiple different tasks without exceeding our limit

Comment (blp) [reply crates/dbsp/src/trace/mod.rs:159]: None of the implementations modified `effort`, and none of the callers read it after calling, so no information was provided or lost.

Comment (Kixiron) [reply crates/dbsp/src/trace/mod.rs:159]: Yet, it's something that's much more beneficial once we start using disk backed traces and cracking

Comment (blp) [reply crates/dbsp/src/trace/mod.rs:159]: Well, if that's a reason to drop this commit, then that's why I named this PR "questionable" :-)

Comment (Kixiron) [reply crates/dbsp/src/trace/mod.rs:159]: The `Activation` stuff I'm not as sure of, that's Leonid's department

**Pull request closed without merging**